### PR TITLE
Implement common hashing for AB#8110

### PR DESCRIPTION
**Pull request by sslaws**
## Status
Ready

## Fixes or Implements [AB#8110](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/8110)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Implemented a common Hash Delegate to be used for protective word caching.

Minor smells corrected.

## Testing
- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### Steps to Reproduce:
N/A

### UI Changes
No

### Browsers Tested
- [ ] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments
N/A

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_404)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_404)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_404)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_404)

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_404)

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_404)

File: Apps/Common/src/Delegates/HMACHashDelegate.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    (KeyDerivationPrf)compareHash.PseudoRandomFunction,
+                    compareHash.Iterations);
+                result = compareHash.Hash == keyHash.Hash;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Generatea a secure random salt of the specified length in bytes.
```
Comment (FuriousLlama) [inline Apps/Common/src/Delegates/HMACHashDelegate.cs]: nit, Generatea -> Generate

Comment (sslaws) [reply Apps/Common/src/Delegates/HMACHashDelegate.cs]: I'll fix this post merge for my next PR.

**Pull request merged**